ddl/ddl_api.go

```diff
@@ -5165,6 +5166,55 @@ func (d *ddl) AlterIndexVisibility(ctx sessionctx.Context, ident ast.Ident, inde
 	return errors.Trace(err)
 }
 
+func checkPlacementSpecConstraint(label string) (placement.LabelConstraint, error) {
+	r := placement.LabelConstraint{}
+
+	if len(label) < 4 {
+		return r, errors.Errorf("constraint too short to be valid: %s", label)
```

Comment:
Use an error code for all placement spec errors, including constraints, role, count.